File: backend/src/main/java/ma/zs/alc/service/facade/collaborator/courseref/LevelTestConfigurationCollaboratorService.java
package ma.zs.alc.service.facade.collaborator.courseref;

import java.util.List;
import ma.zs.alc.bean.core.courseref.LevelTestConfiguration;
import ma.zs.alc.dao.criteria.core.courseref.LevelTestConfigurationCriteria;
import ma.zs.alc.zynerator.service.IService;



public interface LevelTestConfigurationCollaboratorService extends  IService<LevelTestConfiguration,LevelTestConfigurationCriteria>  {

    List<LevelTestConfiguration> findByParcoursId(Long id);
    int deleteByParcoursId(Long id);
    long countByParcoursCode(String code);




}

File: frontend-bqckup/src/app/shared/service/collaborator/courseref/LevelTestConfigurationCollaborator.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';

import {environment} from '../../../../../environments/environment';

import {LevelTestConfigurationDto} from '../../../model/courseref/LevelTestConfiguration.model';
import {LevelTestConfigurationCriteria} from '../../../criteria/courseref/LevelTestConfigurationCriteria.model';
import {AbstractService} from "../../../../zynerator/service/AbstractService";


@Injectable({
    providedIn: 'root'
})
export class LevelTestConfigurationCollaboratorService extends AbstractService<LevelTestConfigurationDto, LevelTestConfigurationCriteria> {
    constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    get API() {
        return environment.apiUrlAlcservice + 'collaborator/levelTestConfiguration/';
    }

    public constrcutDto(): LevelTestConfigurationDto {
        return new LevelTestConfigurationDto();
    }

    public constrcutCriteria(): LevelTestConfigurationCriteria {
        return new LevelTestConfigurationCriteria();
    }
}
